subsys/bluetooth/mesh/settings.c

```diff
@@ -723,18 +723,18 @@ static int va_set(const char *name, size_t len_rd,
 		return 0;
 	}
 
-	if (index >= CONFIG_BT_MESH_LABEL_COUNT) {
-		BT_WARN("Insufficient numbers of lables");
+	lab = get_label(index);
+	if (lab == NULL) {
+		BT_WARN("Out of labels buffers");
 		return -ENOBUFS;
 	}
 
-	lb = get_label();
-	memcpy(lb[index].uuid, va.uuid, 16);
-	lb[index].addr = va.addr;
-	lb[index].ref = va.ref;
+	memcpy(lab[index].uuid, va.uuid, 16);
+	lab[index].addr = va.addr;
+	lab[index].ref = va.ref;
 
 	BT_DBG("Restored Virtual Address, addr 0x%04x ref 0x%04x",
-	       lb[index].addr, lb[index].ref);
+	       lab[index].addr, lab[index].ref);
```

Comment:
should not use the index anymore, as we're getting the pointer directly to the label. All of the `lab[index].` in this function should be `lab->`